libraries/src/MVC/Model/AdminModel.php

```diff
@@ -487,6 +494,21 @@ protected function batchCopy($value, $pks, $contexts)
 			// Get the new item ID
 			$newId = $this->table->get('id');
 
+			if (!empty($oldAssetId))
+			{
+				// Copy rules
+				$query = $db->getQuery(true);
+				$query->clear()
+					->update($db->quoteName('#__assets', 't'))
+					->join('INNER', $db->quoteName('#__assets', 's') .
+						' ON ' . $db->quoteName('s.id') . ' = ' . $oldAssetId
```

Comment:
$oldAssetId should be a prepared parameter
```suggestion
						' ON ' . $db->quoteName('s.id') . ' = :oldAssetId1'
```

Reply:
even in 3.x ?

Reply:
ah... sorry you are right, missed that it's J3... forget my comments